pkg/ingest/ingest.go

```diff
@@ -0,0 +1,101 @@
+package ingest
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+
+	"github.com/thanos-community/obslytics/pkg/dataframe"
+	"github.com/thanos-community/obslytics/pkg/output"
+)
+
+type Ingestor interface {
+	Ingest(*storepb.Series) error
+}
+
+// Options for a single aggregation.
+type AggrOption struct {
+	// Should the aggregation be used?
+	Enabled bool
+	// Column to store the aggregation at
+	Column string
+}
+
+// Collections of aggregations-related options. Determines what aggregations are enabled etc..
+type aggrsOptions struct {
+	// If true (default), the samples should repeat the same time for every day.
+	// This puts additional requirement for 24h to be divisible by the resolution.
+	MidnightAligned bool
+
+	Sum   AggrOption
+	Count AggrOption
+	Min   AggrOption
+	Max   AggrOption
+}
+
+// By default, all aggregations are disabled and target columns set with `_` prefix.
+func NewAggregationsOptions() (ao aggrsOptions) {
+	ao.MidnightAligned = true
```

Comment:
What's this?

Reply:
Also BTW good practice is to avoid boolean option - it's not very extensible. Maybe some `alignment time,Time` would do?

Reply:
See the comment in the attribute itself: the idea is for the data exported over multiple days to be sampled always as the same time of the day: it doesn't matter much for single metric, be I think it would be useful when joining multiple metrics together, when with this, we could use `sample_time` as one of the columns to merge on. This way each metrics could be exported separately but given the same resolution, they would still align nicely.

Regarding boolean, I wanted the aggregator to figure out what the base time would be, rather than getting it from the user of the code… I would suggest leaving it for now until some end2end is in place to see where we could turn it.

Reply:
Turned this option into `initSampleTimeFunc`, by default using `time.Truncate(resolution)`, which seems to do exactly what I needed: having by default deterministic samples boundaries.